crypto/crypto-derive/src/unions.rs

```diff
@@ -8,15 +8,14 @@ use syn::{DataEnum, Ident};
 pub fn parse_newtype_fields(item: &syn::DeriveInput) -> (syn::Type, proc_macro2::TokenStream) {
     let fields = match item.data {
         syn::Data::Struct(ref body) => body.fields.iter().collect::<Vec<&syn::Field>>(),
-        _ => panic!("#[derive(Deref)] can only be used on structs"),
+        _ => vec![],
```

Comment:
As discussed, this should be replaced with [`compile_error!`](https://doc.rust-lang.org/std/macro.compile_error.html) if possible, and reverted to a panic otherwise.

Reply:
By returning an empty vector here the code proceeds to line 18, where a slightly better version of the error message is returned.

I am not trying to rewrite this code in the latest greatest style. I just trying to make things a bit more obvious so that MIRAI knows not to complain about it.

Reply:
Yep, fair enough, my bad.